### PR TITLE
Manual tester post 1

### PR DESCRIPTION
**Pull request by ABizzinotto**
Blog post ready for review. 

Would appreciate your feedback @etagwerker @bronzdoc :)

### CODE REVIEW

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title:  "Manual Testing: How to become a better tester of your own code"
+date: 2019-10-4 11:00:00
+categories: ["software-development", "software-quality"]
+author: abizzinotto
+---
+
+Manual testing is a necessary part of software development and quality assurance. And although it's important to have a dedicated tester in your team, you as a developer can also help speed up the QA, and thus the software development process, by becoming a better manual tester of your own code.
```
Comment (etagwerker) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]:
```suggestion
Manual testing is a necessary part of software development and quality assurance. And although it's important to have a dedicated tester in your team, you as a developer can also help speed up QA, and thus the software development process, by becoming a better manual tester of your own code.
```

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+But how to do that? I'll cover 4 simple points that will help you get there!
+
+<!--more-->
+
+Manual testing is a necessary part of software development and quality assurance. It’s true that it is prone to errors, after all, every human activity is, and can also be time-consuming and boring. It’s also true that automated testing reduces testing time a great deal and is more reliable. But automated tests can only go so far. Human intuition and curiosity cannot be replaced by an automated task.
```
Comment (etagwerker) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: 👍

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Therefore, it is important to have a software tester in your team who knows not only the application but also the end user well. This person will be the “official” manual tester and will find those edge cases for you. But, as a developer, it is also important for you to stay on top of things and manual test your own code before moving it to Quality Assurance. This way, you can ensure your code exactly what it was intended to do.
```
Comment (etagwerker) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]:
```suggestion
Therefore, it is important to have a software tester in your team who knows not only the application but also the end user well. This person will be the “official” manual tester and will find those edge cases for you. But, as a developer, it is also important for you to stay on top of things and manually test your own code before moving on to the Quality Assurance phase. This way, you can ensure your code does exactly what it was intended to do.
```

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Here at OmbuLabs we follow some principles and guidelines to manual test our own code before it goes to QA. In this post, we’ll walk you through these.
```
Comment (etagwerker) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]:
```suggestion
Here at [OmbuLabs](https://www.ombulabs.com) we follow some principles and guidelines to manually test our own code before it goes to QA. In this post, we’ll walk you through these guidelines.
```

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+##It all starts with the user story
+
+Here at OmbuLabs we work with user stories. And every user story has a definition and a scope. Therefore, we have a bug that needs to be fixed, a feature that needs to be added, a chore that needs to be performed, and other situations specific to the project. What all of these have in common is they present a problem to be solved. As such, there is a specific result or outcome that is expected. 
+
+Now, we have automated tests in place to make sure a feature does what it needs and produces the expected outcome. But there are aspects that need manual testing. How does it look on the screen? What if the user decides to follow an alternate path to get the same result?
+
+When manual testing your own code, always go back to the user story and go through the intended flow. Does everything work as expected? Do you see exactly what you’re supposed to see? Is it easy enough for a non-technical person to follow those steps? Where could errors happen?
```
Comment (etagwerker) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]:
```suggestion
When manually testing your own code, always go back to the user story and go through the intended flow. Does everything work as expected? Do you see exactly what you’re supposed to see? Is it easy enough for a non-technical person to follow those steps? Where could errors happen? What are the edge cases that a non-technical user could try? 
```

Not sure if you are interested in this, but it is so true: https://twitter.com/brenankeller/status/1068615953989087232?s=20

Comment (ABizzinotto) [reply _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: Hahaha it's very true! Blog post 2 is about becoming a better manual tester for your team. Maybe I can use this there? I'm not sure how copyright would work in this case though.

Comment (etagwerker) [reply _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: @ABizzinotto I think you can just embed the tweet in the article which takes care of showing proper credit. 😄

Comment (ABizzinotto) [reply _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: @etagwerker awesome! Just did :) I also commited your suggestions to the article. Is there anything else I need to do here?

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Here at [OmbuLabs](https://www.ombulabs.com) we follow some principles and guidelines to manually test our own code before it goes to QA. In this post, we’ll walk you through these guidelines.
+
+##It all starts with the user story
+
+Here at OmbuLabs we work with user stories. And every user story has a definition and a scope. Therefore, we have a bug that needs to be fixed, a feature that needs to be added, a chore that needs to be performed, and other situations specific to the project. What all of these have in common is they present a problem to be solved. As such, there is a specific result or outcome that is expected. 
+
+Now, we have automated tests in place to make sure a feature does what it needs and produces the expected outcome. But there are aspects that need manual testing. How does it look on the screen? What if the user decides to follow an alternate path to get the same result?
```
Comment (bronzdoc) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: `we have automated tests in place to make sure a feature does what it needs and produces the expected outcome.`

We also have automated tests for  "bug fixes"

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+When manually testing your own code, always go back to the user story and go through the intended flow. Does everything work as expected? Do you see exactly what you’re supposed to see? Is it easy enough for a non-technical person to follow those steps? Where could errors happen? What are the edge cases that a non-technical user could try? 
+
+##Write manual tests beforehand
+
+It won’t be possible to just think of all the possibilities and note them down. It isn’t even practical. But it’s always a good idea to take some time to think of the actual actions to be taken by the user to use the feature you’re coding or the one impacted by the bug you’re solving.
```
Comment (bronzdoc) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: Maybe mention that we should automate as much as posible the "manual tests we write" and only rely on them if we have constraints that don't let us automate them.

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+This also needs to be done together with the Product Owner. So when defining a story, use the description area to describe manual tests to be run before moving it to QA. This also helps prevent bias from impacting the quality of the tests. If you write manual tests after coding the feature or just test without writing tests, you’ll be unconsciously biased since you’ll be conditioned to follow the same thought process you followed when coding. As as we know, users are very good at not using the software the way it was intended by the developer.
```
Comment (bronzdoc) [inline _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]:
```suggestion
This also needs to be done together with the Product Owner (Client). So when defining a story, use the description area to describe manual tests to be run before moving it to QA. This also helps prevent bias from impacting the quality of the tests. If you write manual tests after coding the feature or just test without writing tests, you’ll be unconsciously biased since you’ll be conditioned to follow the same thought process you followed when coding. As as we know, users are very good at not using the software the way it was intended by the developer.
```

File: _config.yml [hunk 9]
```diff
@@ -92,3 +92,9 @@ authors:
     email: luis@ombulabs.com
     twitter: bronzdoc
     github: bronzdoc
+  abizzinotto:
+    display_name: "Amanda Bizzinotto"
+    gravatar:
```
Comment (lubc) [inline _config.yml]: @ABizzinotto I just added this changes. Can you put your gravatar id (https://en.gravatar.com/) here so you have an avatar like the rest of us?

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown (hunk 6 above)
Comment (ABizzinotto) [reply _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: Feedback addressed and change made here https://github.com/ombulabs/blog/pull/204.

File: _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown (hunk 7 above)
Comment (ABizzinotto) [reply _posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown]: Feedback addressed and change made here https://github.com/ombulabs/blog/pull/204

File: _config.yml (hunk 9 above)
Comment (lubc) [reply _config.yml]: @ABizzinotto Once you address this comment this will be ready to go  🚀

Comment (ABizzinotto) [reply _config.yml]: @lubc addressed :) Sorry, I had missed this comment.
